constraints/miden-vm/hash.air

```diff
@@ -0,0 +1,156 @@
+mod HashChipletAir
+
+### Constants and periodic columns ################################################################
+
+periodic_columns:
+    k0: [0, 0, 0, 0, 0, 0, 0, 1]
+    k1: [0, 0, 0, 0, 0, 0, 1, 0]
+    k2: [1, 0, 0, 0, 0, 0, 0, 0]
+
+
+### Helper functions/evaluators ###################################################################
+
+# Enforces that column must be binary.
+ev is_binary(main: [a]):
+    enf a^2 - a = 0
+
+
+# Returns binary negation of the value.
+fn binary_not(value: scalar) -> scalar:
+    return 1 - value
+
+
+# Enforces that value in column is copied over to the nex row.
+ev is_copied(main: [column]):
+    ev column' = column
+
+
+# Enforce selector columns constraints
+ev selector_columns(main: [s[3]]):
+    # Set to 1 when either f_hout = 1 or f_sout = 1 in the current row.
+    let f_out = k0 * binary_not(s[0]) * binary_not(s[1])
+    # Set to 1 when either f_hout = 1 or f_sout = 1 in the next row.
```

Comment:
`f_hout` and `f_sout` are undefined.

nit: I'd add a blank line. I think by convention we should have a blank line between a constraint and if the next line is a comment. Wdyt?

Reply:
Yes, I think we should always do this. I left a similar comment previously